hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -1421,6 +1431,53 @@ public void recoverFile(OmKeyArgs keyArgs) throws IOException {
     ozoneClient.getProxy().getOzoneManagerClient().recoverKey(keyArgs, 0L);
   }
 
+  @Override
+  public long finalizeBlock(OmKeyLocationInfo block) throws IOException {
+    incrementCounter(Statistic.INVOCATION_FINALIZE_BLOCK, 1);
```

Comment:
this method appears exactly the same as BasicOzoneClientAdapterImpl.fiinalizeBlock().

Reply:
Yes, most functions' implementation in BasicOzoneClientAdapterImpl and BasicRootedOzoneClientAdapterImpl are the same, one for BasicOzoneFileSystem, one for BasicRootedOzoneFileSystem.

Reply:
There is chance that the whole BasicOzoneClientAdapterImpl and BasicRootedOzoneClientAdapterImpl could be refactored to remove the duplicated codes.